src/Templates/src/templates/maui-blazor/MauiApp.1.csproj

```diff
@@ -52,4 +60,41 @@
         <PackageReference Include="Microsoft.Extensions.Logging.Debug" Version="MS_EXT_VERSION" />
     </ItemGroup>
 
+    <!-- Notes for Publishing on the Mac App Store:
+    1. For macOS it's possible to sign both the app bundle (EnableCodesigning=true) and the package (*.pkg) (EnablePackageSigning=true),
+       and these are signed separately and with different certificates.
+          CodesignKey: this is the signing key used for the app bundle
+          PackageSigningKey: this is the signing key used for the package
+    2. Publishing to the App Store requires signing both the app bundle and the package.
+          Must be 'Apple Distribution: ...' for the app bundle. Note that apps signed like this will not execute locally.
+          They have to be published to the App Store and then downloaded (Apple will resign the app with a different signing identity that allows for local execution).
+          Must be '3rd Party Mac Developer Installer: ...' for the pkg
+    3. Publishing outside of the App Store (i.e. only notarizing) requires:
+          Must be 'Developer ID Application: ...' for the app bundle
+          Must be 'Developer ID Installer: ...' for the pkg
+    4. During development, use the 'Apple Development: ...' signing key (typically to verify that the app works when is signed and entitlements are enforced).
+    5. Depending on the entitlements the app needs, a specific provisioning profile (CodesignProvision) might be needed.
+    6. UseHardenedRuntime must be set to true when app sandbox is enabled in Info.plist.
+    -->
+  <PropertyGroup Condition="$(TargetFramework.Contains('-maccatalyst')) and '$(Configuration)' == 'Debug'">
+    <CodesignEntitlements>Platforms/MacCatalyst/Entitlements.Debug.plist</CodesignEntitlements>
+    <!--
+    <CodesignProvision>YOUR PROFILE NAME</CodesignProvision>
+    <CodesignKey>Apple Development: YOURNAME (*******)</CodesignKey>
+    -->
+  </PropertyGroup>
+
+  <PropertyGroup Condition="$(TargetFramework.Contains('-maccatalyst')) and '$(Configuration)' == 'Release'">
+    <EnableCodeSigning>True</EnableCodeSigning>
+    <ProvisionType>Manual</ProvisionType>
+    <CreatePackage>true</CreatePackage>
+    <EnablePackageSigning>true</EnablePackageSigning>
+    <CodesignEntitlements>Platforms/MacCatalyst/Entitlements.Release.plist</CodesignEntitlements>
+    <UseHardenedRuntime>true</UseHardenedRuntime>
```

Comment:
This is probably why the test is failing. Should we enable this or just add some instructions?
First, this will complicate our pipeline and we need to manually add valid certs each year etc.

Second, this is not something people have today and I think the tooling will now already help get some of this in there? Or at the very least the documentation is there... But not sure what the best option is to do this.